app/controllers/result_filters/location_controller.rb

```diff
@@ -0,0 +1,22 @@
+module ResultFilters
+  class LocationController < ApplicationController
+    include FilterParameters
+
+    def new; end
+
+    def create
+      if(!params[:l])
+        flash[:error] = "Please choose an option"
+        redirect_to location_path(params_without_unsupported_location_params)
+      else
+        redirect_to results_path(params_without_unsupported_location_params)
+      end
+    end
+
+  private
+
+    def params_without_unsupported_location_params
+      filter_params.except(:lat, :lng, :rad, :query, :loc, :lq)
+    end
```

Comment:
Would it be worth it adding support for this to `FilterParams`? For example, `FilterParams.filter_params` does this:

```ruby
    custom_params = request.request_parameters.reject do |param|
      param.in? %w(utf8 authenticity_token)
    end
```

So what about doing this:

```ruby
    custom_params = request.request_parameters.reject do |param|
      param.in? %w(utf8 authenticity_token) + unsupported_params
    end
```

Where `unsupported_params` is defined in this class as `%i[lat lng rad query loc lq]`?

Also, if we decide to change `FilterParams.filter_params`, we can be good boy-scouts and use `except` like the above.

```ruby
    custom_params = request.request_parameters.except(%i(utf8 authenticity_token) + unsupported_params)
```

Reply:
I have a feeling this code will change in the subsequent updates and be in this controller still - lat/lng/rad (and maybe a couple others) will be needed to change in the Postcode ticket, and `lq` in the provider name one will need to change in the provider ticket. So I'll hold off on changing this for now